docs/source/data/data_catalog_yaml_examples.md

```diff
@@ -0,0 +1,395 @@
+```
+```yaml
+cool_dataset:
+  type: pandas.CSVDataSet
+  filepath: "sftp:///path/to/remote_cluster/cool_data.csv"
+  credentials: cluster_credentials
+```
+All parameters required to establish the SFTP connection can be defined through `fs_args` or in the `credentials.yml` file as follows:
+
+```yaml
+cluster_credentials:
+  username: my_username
+  host: host_address
+  port: 22
+  password: password
+```
+The list of all available parameters is given in the [Paramiko documentation](https://docs.paramiko.org/en/2.4/api/client.html#paramiko.client.SSHClient.connect).
+
+## Load multiple datasets with similar configuration using YAML anchors
+
+Different datasets might use the same file format, load and save arguments, and be stored in the same folder. [YAML has a built-in syntax](https://yaml.org/spec/1.2.1/#Syntax) for factorising parts of a YAML file, which means that you can decide what is generalisable across your datasets, so that you need not spend time copying and pasting dataset configurations in the `catalog.yml` file.
+
+You can see this in the following example:
+
+```yaml
+_csv: &csv
+  type: spark.SparkDataSet
+  file_format: csv
+  load_args:
+    sep: ','
+    na_values: ['#NA', NA]
+    header: True
+    inferSchema: False
+
+cars:
+  <<: *csv
+  filepath: s3a://data/01_raw/cars.csv
+
+trucks:
+  <<: *csv
+  filepath: s3a://data/01_raw/trucks.csv
+
+bikes:
+  <<: *csv
+  filepath: s3a://data/01_raw/bikes.csv
+  load_args:
+    header: False
+```
+
+The syntax `&csv` names the following block `csv` and the syntax `<<: *csv` inserts the contents of the block named `csv`. Locally declared keys entirely override inserted ones as seen in `bikes`.
+
+```{note}
+It's important that the name of the template entry starts with a `_` so Kedro knows not to try and instantiate it as a dataset.
+```
+
+You can also nest reuseable YAML syntax:
+
+```yaml
+_csv: &csv
+  type: spark.SparkDataSet
+  file_format: csv
+  load_args: &csv_load_args
+    header: True
+    inferSchema: False
+
+airplanes:
+  <<: *csv
+  filepath: s3a://data/01_raw/airplanes.csv
+  load_args:
+    <<: *csv_load_args
+    sep: ;
+```
+
+In this example, the default `csv` configuration is inserted into `airplanes` and then the `load_args` block is overridden. Normally, that would replace the whole dictionary. In order to extend `load_args`, the defaults for that block are then re-inserted.
+
+## Create a Data Catalog YAML configuration file via the CLI
+
+You can use the [`kedro catalog create` command to create a Data Catalog YAML configuration](../development/commands_reference.md#create-a-data-catalog-yaml-configuration-file).
+
+This creates a `<conf_root>/<env>/catalog/<pipeline_name>.yml` configuration file with `MemoryDataSet` datasets for each dataset in a registered pipeline if it is missing from the `DataCatalog`.
+
+```yaml
+# <conf_root>/<env>/catalog/<pipeline_name>.yml
+rockets:
+  type: MemoryDataSet
+scooters:
+  type: MemoryDataSet
+```
```

Comment:
I put this section in because there was nowhere obvious else to put it but I wonder if it's valid in the YAML examples section. Also, is it valid?

Reply:
A query for @merelcht or @noklam

Reply:
The docs are outdate but should addressed in https://github.com/kedro-org/kedro/issues/2918